[linhv] build failure

**Host Environment**
- OS: Windows
- Compiler: gcc version 11.3.0 (Rev1, Built by MSYS2 project)

**To Reproduce**
1. add vcpkg.json
```json
{
  "$project": "",
  "name": "prescan-plus",
  "version": "0.0.1",
  "dependencies": [
    "libhv"
  ]
}
```
2. add cmakelist
```
cmake_minimum_required(VERSION 3.21)
project(prescanRebuild)
include(E:/vcpkg/scripts/buildsystems/vcpkg.cmake)

set(CMAKE_CXX_STANDARD 17)
find_package(libhv REQUIRED)

add_executable(prescanRebuild main.cpp)

```
3. `vcpkg install --triplet=x64-mingw-dynamic`
4. reload cmake project

**Failure logs**
```
-- Running vcpkg install - done
CMake Error at E:/vcpkg/scripts/buildsystems/vcpkg.cmake:816 (_find_package):
  By not providing "Findlibhv.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "libhv", but
  CMake did not find one.

  Could not find a package configuration file provided by "libhv" with any of
  the following names:

    libhvConfig.cmake
    libhv-config.cmake

  Add the installation prefix of "libhv" to CMAKE_PREFIX_PATH or set
  "libhv_DIR" to a directory containing one of the above files.  If "libhv"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:6 (find_package)
```


I just learned c++, and I'm not sure if the error is due to vcpkg

--- a/ports/libhv/fix-export_cmake.patch
+++ b/ports/libhv/fix-export_cmake.patch
@@ -1,0 +1,37 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f478fb4..5a54eee 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -196,9 +196,9 @@ list_source_directories(LIBHV_SRCS ${LIBHV_SRCDIRS})
+ if(BUILD_SHARED)
+     add_library(hv SHARED ${LIBHV_SRCS})
+     target_compile_definitions(hv PRIVATE HV_DYNAMICLIB)
+-    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS})
++    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
+     target_link_libraries(hv ${LIBS})
+-    install(TARGETS hv
++    install(TARGETS hv EXPORT unofficial-hv-config
+         ARCHIVE DESTINATION lib
+         LIBRARY DESTINATION lib
+         RUNTIME DESTINATION bin)
+@@ -208,12 +208,18 @@ endif()
+ if(BUILD_STATIC)
+     add_library(hv_static STATIC ${LIBHV_SRCS})
+     target_compile_definitions(hv_static PRIVATE HV_STATICLIB)
+-    target_include_directories(hv_static PRIVATE ${LIBHV_SRCDIRS})
++    target_include_directories(hv_static PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
+     target_link_libraries(hv_static ${LIBS})
+-    install(TARGETS hv_static DESTINATION lib)
++    install(TARGETS hv_static EXPORT unofficial-hv-config DESTINATION lib)
+     add_custom_target(libhv_static DEPENDS hv_static)
+ endif()
+ 
++INSTALL(EXPORT unofficial-hv-config
++    NAMESPACE unofficial::hv::
++    DESTINATION share/unofficial-hv
++    FILE unofficial-hv-config.cmake
++)
++
+ file(INSTALL ${LIBHV_HEADERS} DESTINATION include/hv)
+ install(FILES ${LIBHV_HEADERS} DESTINATION include/hv)
+ 

--- a/ports/libhv/fix-export_cmake.patch
+++ b/ports/libhv/fix-export_cmake.patch
@@ -1,20 +1,20 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index f478fb4..5a54eee 100644
+index f478fb4..1e3556a 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -196,9 +196,9 @@ list_source_directories(LIBHV_SRCS ${LIBHV_SRCDIRS})
+@@ -196,9 +196,10 @@ list_source_directories(LIBHV_SRCS ${LIBHV_SRCDIRS})
  if(BUILD_SHARED)
      add_library(hv SHARED ${LIBHV_SRCS})
      target_compile_definitions(hv PRIVATE HV_DYNAMICLIB)
 -    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS})
 +    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
      target_link_libraries(hv ${LIBS})
--    install(TARGETS hv
-+    install(TARGETS hv EXPORT unofficial-hv-config
+     install(TARGETS hv
++        EXPORT libhvConfig
          ARCHIVE DESTINATION lib
          LIBRARY DESTINATION lib
          RUNTIME DESTINATION bin)
-@@ -208,12 +208,18 @@ endif()
+@@ -208,14 +209,17 @@ endif()
  if(BUILD_STATIC)
      add_library(hv_static STATIC ${LIBHV_SRCS})
      target_compile_definitions(hv_static PRIVATE HV_STATICLIB)
@@ -22,16 +22,15 @@ index f478fb4..5a54eee 100644
 +    target_include_directories(hv_static PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
      target_link_libraries(hv_static ${LIBS})
 -    install(TARGETS hv_static DESTINATION lib)
-+    install(TARGETS hv_static EXPORT unofficial-hv-config DESTINATION lib)
++    install(TARGETS hv_static
++        EXPORT libhvConfig
++        ARCHIVE DESTINATION lib)
      add_custom_target(libhv_static DEPENDS hv_static)
  endif()
  
-+INSTALL(EXPORT unofficial-hv-config
-+    NAMESPACE unofficial::hv::
-+    DESTINATION share/unofficial-hv
-+    FILE unofficial-hv-config.cmake
-+)
-+
  file(INSTALL ${LIBHV_HEADERS} DESTINATION include/hv)
  install(FILES ${LIBHV_HEADERS} DESTINATION include/hv)
++install(EXPORT libhvConfig DESTINATION lib/cmake/libhv)
  
+ if(BUILD_SHARED)
+     set(HV_LIBRARIES hv CACHE INTERNAL "link hv libraries")

--- a/ports/libhv/portfile.cmake
+++ b/ports/libhv/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-find_package.patch
+        fix-export_cmake.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
@@ -29,6 +30,7 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-hv CONFIG_PATH share/unofficial-hv)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/libhv/portfile.cmake
+++ b/ports/libhv/portfile.cmake
@@ -6,7 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-find_package.patch
-        fix-export_cmake.patch
+        fix-export_cmake.patch #Sync up the upstream changes
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
@@ -30,7 +30,7 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-hv CONFIG_PATH share/unofficial-hv)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libhv)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/libhv/vcpkg.json
+++ b/ports/libhv/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libhv",
   "version": "1.2.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Libhv is a C/C++ network library similar to libevent/libuv.",
   "homepage": "https://github.com/ithewei/libhv",
   "license": "BSD-3-Clause",
@@ -9,6 +9,10 @@
   "dependencies": [
     {
       "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
       "host": true
     }
   ],

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3658,7 +3658,7 @@
     },
     "libhv": {
       "baseline": "1.2.4",
-      "port-version": 1
+      "port-version": 2
     },
     "libhydrogen": {
       "baseline": "2021-12-02",

--- a/versions/l-/libhv.json
+++ b/versions/l-/libhv.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "7ef2e1cb3639572bed7cbaf78a46659a6f34fc59",
+      "git-tree": "3533f306ae38ad29c619656d006ae9536bfcdcb4",
       "version": "1.2.4",
       "port-version": 2
     },

--- a/versions/l-/libhv.json
+++ b/versions/l-/libhv.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7ef2e1cb3639572bed7cbaf78a46659a6f34fc59",
+      "version": "1.2.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "b5953fb3634520fafd04507d0a7266e92c87a129",
       "version": "1.2.4",
       "port-version": 1

